portfolio/src/main/java/com/google/sps/servlets/ProximityPinsServlet.java

```diff
@@ -0,0 +1,75 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.*;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/proximity-pins")
+public class ProximityPinsServlet extends HttpServlet {
+  private static final String STREET_ADDRESS = "address";
+  private static final String CITY = "city";
+  private static final String STATE = "state";
+  private static final String ZIPCODE = "zipcode";
+  private static final String RACE = "race";
+  private static final String CAUSE_OF_DEATH = "cause of death";
+  private static final String DATE_OF_DEATH = "date";
+  private static final String TABLE_NAME = "Police Violence Data";
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    String queryState = request.getParameter(STATE);
+    Filter stateFilter = new FilterPredicate(STATE, FilterOperator.EQUAL, queryState);
+    Query query = new Query(TABLE_NAME).setFilter(stateFilter);
+    PreparedQuery results = datastore.prepare(query);
+    response.setContentType("application/json");
+
+    List<ProximityPin> quieriedPins = new ArrayList<ProximityPin>();
+    for (Entity entity : results.asIterable()) {
+      String streetAddress = (String) entity.getProperty(STREET_ADDRESS);
+      String city = (String) entity.getProperty(CITY);
+      String state = (String) entity.getProperty(STATE);
+      String zipcode = (String) entity.getProperty(ZIPCODE);
+      String race = (String) entity.getProperty(RACE);
+      String causeOfDeath = (String) entity.getProperty(CAUSE_OF_DEATH);
+      String dateOfDeath = (String) entity.getProperty(DATE_OF_DEATH);
+      ProximityPin proximityPin =
+          new ProximityPin(streetAddress, city, state, zipcode, race, causeOfDeath, dateOfDeath);
+      quieriedPins.add(proximityPin);
```

Comment:
nit: queried is spelled wrong here too. just don't forget to change the spelling at every use